library/ssl_cli.c

```diff
@@ -2065,6 +2055,29 @@ static int ssl_parse_hello_verify_request( mbedtls_ssl_context *ssl )
 }
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
 
+static int is_compression_ok( mbedtls_ssl_context *ssl, unsigned char comp )
+{
+    int accept_comp = 1;
```

Comment:
Minor:
I'm not a big fan of this logic reversal.
With this PR - if there's no compression support, we still "accept" compression.
If we changed this variable to "is_comp_bad" and function to "is_compression_bad" it would make sense that if compression is not available, it can't be bad. You would then just check for `is_compression_bad` instead of `!is_compression_ok` in the place that calls it.
That being said, it's mostly a matter of preference.